src/App.js

```diff
@@ -1,85 +1,356 @@
-
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main">
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">asdfghj</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">qwertyuio</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">zxcvbnm</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-4" />
-              <label htmlFor="todo-4">1234567890</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+class App extends React.Component {
+  state = {
+    todos: [],
+    nextId: 1,
+    typeOfFilter: 'all',
+    isAllButtonActive: true,
+    isActiveButtonActive: false,
+    isComplitedButtonActive: false,
+    isToggleAll: true,
+    fieldValue: '',
+    isAnyEditedField: false,
+  }
+
+  componentDidMount() {
+    if (!localStorage.getItem('ToDoAppData')) {
+      return;
+    }
+
+    const initState = JSON.parse(localStorage.getItem('ToDoAppData'));
+    const {
+      todos,
+      nextId,
+      typeOfFilter,
+      isAllButtonActive,
+      isActiveButtonActive,
+      isComplitedButtonActiv,
+      isToggleAll,
+      fieldValue,
+    } = initState;
+
+    this.setState({
+      todos,
+      nextId,
+      typeOfFilter,
+      isAllButtonActive,
+      isActiveButtonActive,
+      isComplitedButtonActiv,
+      isToggleAll,
+      fieldValue,
+    });
+  }
+
+  componentDidUpdate() {
+    localStorage.setItem('ToDoAppData', JSON.stringify(this.state));
+  }
+
+  handleIsActiveChange = (event) => {
+    // const { todos } = this.state;
+    //   this.setState(prevState => ({
+    //     isToggleAll: !prevState.isToggleAll,
+    //   }));
+    // }
+    const { id } = event.target.parentElement.parentElement;
```

Comment:
you are interacting with dom again

Reply:
Don't know how to get {id} in different way.